src/js/controller/model.js

```diff
@@ -102,7 +102,7 @@ define([
 
                     /* falls through */
                 case events.JWPLAYER_MEDIA_META:
-                    var duration = data.duration;
+                    var duration = data.duration || 0;
```

Comment:
@pajong @egreaves The provider changes I made below are where the duration is potentially being set to NaN in certain race conditions due to the browser. This is the common location where all providers pass through to set the duration, so I made this change to always ensure we have a default of 0 for invalid values. This is the same way we initialize the duration value to begin with, using `|| 0` as a fallback value.